tests/Integration/Behaviour/Features/Context/Domain/ShipmentFeatureContext.php

```diff
@@ -30,14 +30,57 @@
 namespace Tests\Integration\Behaviour\Features\Context\Domain;
 
 use Behat\Gherkin\Node\TableNode;
+use Exception;
 use PHPUnit\Framework\Assert;
+use PrestaShop\PrestaShop\Core\Domain\Shipment\Command\SwitchShipmentCarrierCommand;
 use PrestaShop\PrestaShop\Core\Domain\Shipment\Query\GetOrderShipments;
 use PrestaShop\PrestaShop\Core\Domain\Shipment\Query\GetShipmentProducts;
 use RuntimeException;
 use Tests\Integration\Behaviour\Features\Context\SharedStorage;
 
 class ShipmentFeatureContext extends AbstractDomainFeatureContext
 {
+    /**
+     * @When I switch the carrier for shipment :shipmentReference to :carrierReference
+     */
+    public function switchShipmentCarrier(string $shipmentReference, string $carrierReference): void
+    {
+        $shipmentId = SharedStorage::getStorage()->get($shipmentReference);
+        $carrierId = $this->referenceToId($carrierReference);
+
+        try {
+            $this->getCommandBus()->handle(new SwitchShipmentCarrierCommand($shipmentId, $carrierId));
+        } catch (Exception $error) {
+            throw new RuntimeException(sprintf('Error then switch shipment "%s" to carrier "%s" : %s', $shipmentReference, $carrierReference, $error->getMessage()));
```

Comment:
Is that what you meant?
```suggestion
            throw new RuntimeException(sprintf('Error while switching shipment "%s" to carrier "%s" : %s', $shipmentReference, $carrierReference, $error->getMessage()));
```